common/djangoapps/util/course.py

```diff
@@ -2,27 +2,49 @@
 Utility methods related to course
 """
 import logging
-from django.conf import settings
 
+from django.conf import settings
 from opaque_keys.edx.keys import CourseKey
 
+from openedx.core.djangoapps.catalog.utils import get_run_marketing_url
+
 log = logging.getLogger(__name__)
 
 
-def get_lms_link_for_about_page(course_key):
+def get_link_for_about_page(course_key, user):
     """
     Returns the url to the course about page.
     """
     assert isinstance(course_key, CourseKey)
 
     if settings.FEATURES.get('ENABLE_MKTG_SITE'):
-        # Root will be "https://www.edx.org". The complete URL will still not be exactly correct,
-        # but redirects exist from www.edx.org to get to the Drupal course about page URL.
-        about_base = settings.MKTG_URLS['ROOT']
-    else:
-        about_base = settings.LMS_ROOT_URL
+        marketing_url = get_run_marketing_url(course_key, user)
+        if marketing_url:
+            return marketing_url
+
+    return get_lms_course_about_page_url(course_key)
+
+
+def get_link_for_about_page_from_cache(course_key, catalog_course_run=None):
```

Comment:
This method seems very similar to the one above. Is there any way to make this code more DRY?